kubernetes.go

```diff
@@ -106,23 +118,32 @@ func (k *KubernetesRegistry) LookupService(ctx context.Context, name string) (Se
 	for _, obj := range k.store.List() {
 		pod, err := toPod(obj)
 		if err != nil {
-			events.Log("failed to covert data to pod: %{error}s", err)
+			events.Log("failed to convert data to pod: %{error}s", err)
 			continue
 		}
-
-		for _, container := range pod.Spec.Containers {
-			tags := []string{pod.Name}
-
-			for _, port := range container.Ports {
-				hosts = append(hosts, Host{
-					Addr: &net.TCPAddr{
-						IP:   net.ParseIP(pod.Status.PodIP),
-						Port: int(port.ContainerPort),
-					},
-					Tags: append(tags, port.Name),
-				})
+    // filtering pods based on podname, even if they are diff namepsaces for now, since the route for namespaces isnt made yet
+		if pod.Name == name {
+			for _, container := range pod.Spec.Containers {
+				// adding container name to display
+				tags := []string{pod.Name+"-"+container.Name}
+
+				for _, port := range container.Ports {
+					if port.Name == "http" {
+						hosts = append(hosts, Host{
+							Addr: &net.TCPAddr{
+								IP:   net.ParseIP(pod.Status.PodIP),
+								Port: int(port.ContainerPort),
+							},
+							Tags: append(tags, port.Name), // port name must be specified in the pod spec as http
+						})
+					} else {
+						continue
```

Comment:
Should we remove this `else` + `continue`, it doesn't seem to change the flow of execution.